application/src/test/java/be/jpendel/domain/event/EventFactoryTest.java

```diff
@@ -0,0 +1,24 @@
+package be.jpendel.domain.event;
+
+import be.jpendel.application.CreateEventCommand;
+import org.junit.Test;
+
+import java.time.LocalDateTime;
+
+public class EventFactoryTest {
+
+    @Test
+    public void eventWithSameNameAndSameStartdateLocationCanNotBeCreated(){
+       EventFactory factory = new EventFactory();
+        LocalDateTime time = LocalDateTime.now();
+        CreateEventCommand eventCmd = CreateEventCommand.builder()
+                .withEndDateTime(time).withLocation("loc")
+                .withStartDateTime(time)
+                .withName("test").build();
+        Event e1 = factory.create(eventCmd);
+        EventRepository repo = new InMemEventRepository();
+        repo.add(e1);
+        repo.add(e1);// moet dit falen ?
```

Comment:
Idd, maar das dan een beetje het punt, als ge hem wegmocked dan mocked ge echte repo logica weg, dan is dit project een beter idee: https://github.com/mmnaseri/spring-data-mock

Dan houden we onze echte spring data repo's met hun behaviour, wat we misschien wel willen hebben in onze testen, maar zonder echte db.

Er zou trouwens nog een andere optie kunnen zijn dan de factory of de repository en das een BusinessRule hiervoor creeeren. In de usecase wordt er dan gewoon gechecked of er aan voldaan is.

createNewMember(newMemberCommand){
    Member newMember = memberFactory(newMemberCommand);
    usernameUniqueBusinessRule.validate(newMember);
}

en dat die evt een BusinessException smijt met wat er mis is, in dit geval ne messagekey member.with.username.already.exists dat we in de infra layer kunnen vertalen.

We moeten dringend terug beginnen mob programmen om die dingen uit te klaren ;)